src/Data/Managers/V2/PostMetadata/usePostMetadata.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Custom hook for posting metadata.
+ *
+ * @param environment - The environment to use for creating headers.
+ * @returns - The mutation object from `useMutation` hook.
+ */
+import { useMutation } from "@tanstack/react-query";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../helpers";
+
+export const usePostMetadata = (environment: string) => {
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+  const { createHeaders, handleErrors } = useFetchHelpers();
+  const headers = createHeaders(environment);
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
+
+  /**
+   * Posts metadata.
+   *
+   * @param info - The metadata information to post.
+   */
+  const postMetadata = async (info: {
```

Comment:
Can we have a type for this? I think it might be handy to be able to reuse this type when you're using the hook.

Reply:
added